packages/selinux-policy/serge-testsuite/lib/common.sh

```diff
@@ -0,0 +1,268 @@
+        grep -q "^$1 " $STATUSFILE ||  getsebool $1 >> $STATUSFILE
+        # now switch the boolean on
+        setsebool -P $1 on
+        shift
+    done
+
+}
+
+
+# switch the boolean(s) off
+# Usage:  slSEBooleanOff boolean1 [bolean2 ...]
+function rlSEBooleanOff() {
+
+    local STATUSFILE="$BEAKERLIB_DIR/sebooleans" && touch $STATUSFILE
+
+    rlLog "Setting SELinux booleans $* off"
+    while [ -n "$1" ]; do
+        # if we didn't save the status yet, save it now
+        grep -q "^$1 " $STATUSFILE || getsebool $1 >> $STATUSFILE
+        # now switch the boolean on
+        setsebool -P $1 off
+        shift
+    done
+
+}
+
+
+# restore original state of SELinux boolean(s) - all used booleans if no specified
+# Usage rlSEBooleanRestore [boolean1 ...]
+function rlSEBooleanRestore() {
+
+    local STATUSFILE="$BEAKERLIB_DIR/sebooleans"
+    local RECORD
+
+    if [ ! -f $STATUSFILE ]; then
+        rlLogError "Cannot restore SELinux booleans, saved states are not available"
+        return 99
+    fi
+
+    if [ -z "$1" ]; then     # no booleans specified, restoring all booleans
+        rlLog "Restoring all used SELinux booleans"
+        cat $STATUSFILE | while read RECORD; do
+            # restore original boolean status saved in a STATUSFILE
+            setsebool -P `echo $RECORD | cut -d ' ' -f 1` `echo $RECORD | cut -d ' ' -f 3`
+        done
+    else      # restoring only specified booleans
+        rlLog "Restoring original status of SELinux booleans $*"
+        while [ -n "$1" ]; do   # process all passed booleans
+            RECORD=`grep "^$1 " $STATUSFILE`
+            if [ -z "$RECORD" ]; then
+                rlLogError "Cannot restore SELinux boolean $1, original state was not saved"
+            else
+                # restore original boolean status saved in a STATUSFILE
+                setsebool -P `echo $RECORD | cut -d ' ' -f 1` `echo $RECORD | cut -d ' ' -f 3`
+            fi
+            shift
+        done
+    fi
+}
+
+
+# execute given command and whenever prompted for password, provides it
+# password can be stored in $PASSWORD global variable or specified using -p parameter
+# usage: rlSERunWithPassword [ -p PASSWORD ] COMMAND
+function rlSERunWithPassword() {
+
+  local PASS="$PASSWORD"
+  # read password parameter
+  if [ "$1" = "-p" ]; then
+      PASS="$2"
+      shift 2
+  fi
+
+  cat <<EOF | expect -
+set timeout 5
+spawn $*
+expect {
+  "*assword:*" { send "$PASS\r"; exp_continue }
+}
+EOF
+
+}
+
+
+# tests whether MLS policy is used
+function rlSEIsMLS() {
+    sestatus | grep -qi mls
+}
+
+
+# tests whether targeted policy is used
+function rlSEIsTargeted() {
+    sestatus | grep -qi targeted
+}
+
+
+# tests whether strict policy is used
+function rlSEIsStrict() {
+    sestatus | grep -qi strict
+}
+
+
+
+# install requires
+SE_REQUIRES=setools-console
+rlIsRHEL 4 5 && SE_REQUIRES=setools
+rpm -q $SE_REQUIRES || yum -y install $SE_REQUIRES || echo "Error: cannot install $SE_REQUIRES"
```

Comment:
While yum is aliased to dnf on the current systems, it may not be the case later. Would it be possible to check whether the system uses yum or dnf and then use that for installation, like for example here: https://github.com/CKI-project/tests-beaker/blob/master/distribution/kpkginstall/runtest.sh#L163 ?

Reply:
Hi @veruu, I added a function select_yum_tool(), and updated common.sh accordingly, please take a look again, thanks!